tests/roles.test.ts

```diff
@@ -0,0 +1,112 @@
+        environment.setCurrentTime(now);
+
+        // Proposing
+        await environment.dao.contract.propose({
+            proposer: environment.members[0].account.accountName,
+            proposal_type: 'role',
+            ...testRole
+        });
+        
+        let proposal = last(getDocumentsByType(
+            environment.getDaoDocuments(),
+            'role'
+        ));
+
+        getDaoExpect(environment).toHaveEdge(environment.getRoot(), proposal, 'proposal');
+
+        let proposalDetails = getDetailsGroup(proposal);
+
+        await expect(getContent(proposalDetails, "state").value[1])
+              .toBe('proposed');
+        
+        let date = nextDay(environment, new Date());
+
+        // closes proposal
+        await environment.dao.contract.closedocprop({
+          proposal_hash: proposal.hash
+        }, environment.members[0].getPermissions())
+      
+        proposal = last(getDocumentsByType(
+          environment.getDaoDocuments(),
+          'role'
+        ));
+
+        getDaoExpect(environment).toHaveEdge(environment.getRoot(), proposal, 'failedprops');
+
+        proposalDetails = getDetailsGroup(proposal);
+
+        await expect(getContent(proposalDetails, "state").value[1])
+              .toBe('rejected');
+
+        //Propose the role again
+       
+        // Proposing
+        await environment.dao.contract.propose({
+            proposer: environment.members[0].account.accountName,
+            proposal_type: 'role',
+            ...testRole
+        });
+        
+        proposal = last(getDocumentsByType(
+            environment.getDaoDocuments(),
+            'role'
+        ));
+
+        proposalDetails = getDetailsGroup(proposal);
+
+        await expect(getContent(proposalDetails, "state").value[1])
+              .toBe('proposed');
+
+        for (let i = 0; i < environment.members.length; ++i) {
+          console.log("Voting to pass:", i);
+          await environment.dao.contract.vote({
+            voter: environment.members[i].account.accountName,
+            proposal_hash: proposal.hash,
+            vote: 'pass',
+            notes: 'votes pass'
+          });
+        }
+
+        const expiration = getContent(getContentGroupByLabel(proposal, "ballot"), "expiration");
+        
+        date = setDate(environment, new Date(expiration.value[1]), 15);
+
+        // closes proposal
+        await environment.dao.contract.closedocprop({
+          proposal_hash: proposal.hash
+        }, environment.members[0].getPermissions());
+      
+        proposal = last(getDocumentsByType(
+          environment.getDaoDocuments(),
+          'role'
+        ));
+
+        proposalDetails = getDetailsGroup(proposal);
+
+        getDaoExpect(environment).toHaveEdge(environment.getRoot(), proposal, 'passedprops');
+        getDaoExpect(environment).toHaveEdge(environment.getRoot(), proposal, 'role');
+
+        await expect(getContent(proposalDetails, "state").value[1])
```

Comment:
We don't need to await when the methods are sync. (i.e. `expect` on functions that do not return a promise/async)

Reply:
Good to know!